misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -245,10 +256,39 @@ internal class RealTransacter private constructor(
       return session.doReturningWork(work)
     }
 
+    internal fun preCommit() {
+      preCommitHooks.forEach {
+        // Propagate hook exceptions up to the transacter so that the the transaction is rolled
+        // back and the error gets returned to the applicatiob
+        it()
+      }
+    }
+
+    override fun onPreCommit(work: () -> Unit) {
+      preCommitHooks.add(work)
+    }
+
+    internal fun postCommit() {
+      postCommitHooks.forEach {
+        // At this point the transaction has committed, so simply log errors and don't propagate
```

Comment:
I think I would still want the exception to propagate, even if the tx has been committed and can't be rolled back. For example it would be nice for the current WebAction to terminate with a 500 and be allowed to be retried by the caller, hopefully then the postCommit works

Reply:
I don't think I agree. If `Transacter.transaction` throws an exception, most code is going to assume that the transaction failed - which it didn't, and re-running the work in the transaction may not be possible. We could potentially use a specific exception to differentiate these cases (`PostCommitHookFailedException`) but I think this might be too subtle for application code.

Reply:
+1, if there's important work happening in a postCommit, you probably want a heads up that it didn't happen.

Reply:
Ok, I'll wrap in a specific exception.